spanner/value_benchmarks_test.go

```diff
@@ -230,3 +232,167 @@ func decodeArrayReflect(pb *proto3.ListValue, name string, typ *sppb.Type, aptr
 	}
 	return nil
 }
+
+func BenchmarkScan100RowsUsingSelectAll(b *testing.B) {
```

Comment:
It seems that we only have benchmarks for this function, but no unit- and integration tests. Can we add that? Maybe re-use the tests that currently use the method that was deemed to slow to verify that the output is the same?

Reply:
Added unit tests